.github/workflows/tech-debt-burndown.yml

```diff
@@ -0,0 +1,139 @@
+name: Tech Debt Burndown
+# this workflow is for reporting on various metrics for the codebase that
+# we want to pay attention to. Generally these are checks of some sort that we'll
+# want to eventually 'graduate' to full CI checks (which cause builds to fail if
+# there are any errors) once we've eliminated all the problems, but until that
+# point we run them here, separate from the main build, and write a report on our
+# progress on them to each PR.
+#
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  strict_null_check: # TODO(STENCIL-446): Remove this workflow once `strictNullChecks` is enabled
+    strategy:
+      matrix:
+        branch: [ 'main', 'pr' ]
+    name: 'Get strictNullChecks errors on ${{ matrix.branch }}'
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        with:
+          ref: main
+        if: ${{ matrix.branch == 'main' }}
+
+      - name: Checkout PR branch
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        if: ${{ matrix.branch == 'pr' }}
+
+      - name: Get Core Dependencies
+        uses: ./.github/workflows/actions/get-core-dependencies
+
+      - name: Install tsc-output-parser
+        run: npm install @aivenio/tsc-output-parser@2.1.1
+
+      - name: Run Typescript compiler and generate JSON-formatted error file
+        run: npx tsc --strictNullChecks --noEmit --pretty false | npx tsc-output-parser > null_errors_${{ matrix.branch }}.json
+
+      - name: Upload null_errors_${{ matrix.branch }}.json
+        uses: actions/upload-artifact@6673cd052c4cd6fcf4b4e6e60ea986c889389535 # v3.0.0
+        with:
+          name: null_errors_${{ matrix.branch }}
+          path: 'null_errors_${{ matrix.branch }}.json'
+
+  # TODO(STENCIL-454): Remove or change this up once we've eliminated unused exports
+  unused_exports_check:
+    strategy:
+      matrix:
+        branch: [ 'main', 'pr' ]
+    name: Find unused variables on ${{ matrix.branch }}
+    runs-on: 'ubuntu-latest'
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        with:
+          ref: main
+        if: ${{ matrix.branch == 'main' }}
+
+      - name: Checkout PR branch
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+        if: ${{ matrix.branch == 'pr' }}
+
+      - name: Install ts-prune
+        run: npm install ts-prune@0.10.3
+
+      - name: Run ESLint and write output to disk
+        run: npx ts-prune > unused-exports-${{ matrix.branch }}.txt
```

Comment:
Is the name of this step still accurate?

Reply:
it is not 😅